Week3/homework/6-step3.js

```diff
@@ -3,4 +3,25 @@
 const arr2d = [[1, 2], [3, 4], [5, 6]];
 const arr3d = [[[1, 2], [3, 4]], [[5, 6], [7, 8]]];
 
-// add your solution here, or add a comment on how you would tackle this problem
+let arrText2 = '';
+let arrText3 = '';
+
+for (let i = 0; i < arr2d.length; i++) {
+  for (let j = 0; j < arr2d[i].length; j++) {
+    arrText2 += arr2d[i][j] + '   ';
+  }
+  console.log(arrText2);
+  arrText2 = '';
+}
+
+for (let i = 0; i < arr3d.length; i++) {
+  for (let j = 0; j < arr3d[i].length; j++) {
+    arrText3 += arr3d[i][j] + '   ';
+  }
+  console.log(arrText3);
+  arrText3 = '';
+}
+
+/* There are many ways to print out a multiple dimension arrays depending on the purpose of this procedure */
+console.log(arr2d);
+console.log(arr3d);
```

Comment:
The idea here is that you want each item to be printed on its own.  What you are doing is for each element in the loop, checking if it is also an array, and if so, descending into that array, and printing its elements, if any of them is an array, descend. Rinse and repeat until you have traversed the whole tree (nested arrays like this are essentially trees).

As an example here is how I tackled this problem, and a rather nasty mess of nested arrays to test it on.

``` 
const arr2d = [[1, 2], [3, 4], [5, 6]];
const arr3d = [[[1, 2], [3, 4]], [[5, 6], [7, 8]]];
const arrNd = [[[[1, 2]], [3, 4], [[[5, 6], [7, 8]], [9, 10]], [11, 12]], [[[[[[13, 14]]]]]]];

const runArray = (arr) => {
  arr.map((elem) => {
    if (Array.isArray(elem)) {
      return runArray(elem);
    } else {
      console.log(elem);
    }
  });
};

// runArray(arr2d);
// runArray(arr3d);
runArray(arrNd);
```
I hope this helps illustrate (one) way to solve this.  The important part is detecting an array, and then recalling the function with this array.